src/app/pcasts/dao/users_dao.py

```diff
@@ -66,3 +66,23 @@ def search_users(search_name, offset, max_search):
       (User.username.like(search_name+'%')) \
       .offset(offset).limit(max_search).all()
   return possible_users
+
+def change_user_name(old_name, new_name):
+    user = User.query.filter(User.username == old_name).first()
+    if user:
+      if User.query.filter(User.username == new_name).first():
+        raise Exception("Username " + new_name + " already in use")
+      #There is no integrity constraint for empty strings in SQL
```

Comment:
We can check simple constraints like this at the model level, see [this](https://stackoverflow.com/questions/8256715/simple-validation-with-sqlalchemy) for an example of how to write a "validator".